### PR TITLE
Cb 45

### PR DESCRIPTION
**Pull request by CalebMacdonaldBlack**

### CODE REVIEW

File: src/cljc/centrebull/spec.cljc [hunk 1]
```diff
@@ -74,7 +74,7 @@
 ;Clojure spec predicate for a non empty string
 (def non-empty-string (s/and string? #(not= "" %)))
 
-(defn- str->int [s] #?(:cljs (cond (and (string? s) #(not= "" %)) (cljs.tools.reader/read-string s)
+(defn- str->int [s] #?(:cljs (cond (and (string? s) #(not= "" %)) s
```
Comment (CalebMacdonaldBlack) [inline src/cljc/centrebull/spec.cljc]: I was getting `Uncaught TypeError: Cannot read property 'reader' of undefined`. I don't completely understand what this function does but it works with this change

File: src/cljs/centrebull/shooters/view.cljs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  [:section
+   [:card
+    [:h2 {:local "9/12"} "Shooters"]
+    [:button {:local "3/12" :on-click toggle-action} "New Shooter"]
+
+    [search "/shooters/search"
+     (fn [{:keys [shooter/sid shooter/preferred-name shooter/first-name shooter/last-name shooter/club]}]
+       [:div {:on-click #(rf/dispatch [:set-active-shooter sid])}
+        [:div {:local "1/3"} sid]
+        [:div {:local "1/3"} (if (empty? preferred-name) (str first-name " " last-name) preferred-name)]
```
Comment (CalebMacdonaldBlack) [inline src/cljs/centrebull/shooters/view.cljs]: You might not want it done like this

File: src/less/_variables.less [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 // set your brand colour
 
-@primary: #1985A1;
+@primary: indianred;
```
Comment (CalebMacdonaldBlack) [inline src/less/_variables.less:27]: I like this look. What do you think?

File: src/cljc/centrebull/spec.cljc (hunk 1 above)
Comment (axrs) [reply src/cljc/centrebull/spec.cljc]: The cljs.tools/reader reads a string and attempts to convert it into a primative type. i.e. "1234" becomes 1234.

File: src/cljs/centrebull/competitions/core.cljs [hunk 4]
```diff
@@ -2,10 +2,10 @@
   (:require
     [secretary.core :as secretary]
     [centrebull.competitions.handlers]
-    [clojure.spec :as s]
     [centrebull.competitions.views :as v]
     [re-frame.core :as rf]
-    [reagent.core :as r]))
+    [reagent.core :as r]
+    [cljs.spec :as s]))
```
Comment (axrs) [inline src/cljs/centrebull/competitions/core.cljs]: clojure.spec and cljs.spec resolve the same namespace by the compiler

File: src/cljs/centrebull/shooters/view.cljs (hunk 2 above)
Comment (axrs) [reply src/cljs/centrebull/shooters/view.cljs]: That's fine. Mabe move the calculated field of first-name + last-name into a let statement above.

File: src/cljc/centrebull/spec.cljc (hunk 1 above)
Comment (axrs) [reply src/cljc/centrebull/spec.cljc]: Have a go at fixing this. It should just be requiring reader.

**Review by axrs — CHANGES_REQUESTED** [review]

**Comment by CalebMacdonaldBlack** [comment]
@axrs

File: src/cljs/centrebull/shooters/view.cljs [hunk 5]
```diff
@@ -11,10 +11,11 @@
 
     [search "/shooters/search"
      (fn [{:keys [shooter/sid shooter/preferred-name shooter/first-name shooter/last-name shooter/club]}]
-       [:div {:on-click #(rf/dispatch [:set-active-shooter sid])}
-        [:div {:local "1/3"} sid]
-        [:div {:local "1/3"} (if (empty? preferred-name) (str first-name " " last-name) preferred-name)]
-        [:div {:local "1/3"} club]])]]])
+       (let [full-name (str first-name " " last-name)]
```
Comment (axrs) [inline src/cljs/centrebull/shooters/view.cljs]:
```
(let [name (or preferred-name (str first-name " " last-name))]
...
)
```

File: src/cljc/centrebull/spec.cljc [hunk 6]
```diff
@@ -74,7 +74,7 @@
 ;Clojure spec predicate for a non empty string
 (def non-empty-string (s/and string? #(not= "" %)))
 
-(defn- str->int [s] #?(:cljs (cond (and (string? s) #(not= "" %)) s
+(defn- str->int [s] #?(:cljs (cond (and (string? s) #(not= "" %)) (clojure.tools.reader/read-string s)
```
Comment (axrs) [inline src/cljc/centrebull/spec.cljc]: Is clojure.tools.reader available in cljs?

**Review by axrs — CHANGES_REQUESTED** [review]

**Comment by axrs** [comment]
@CalebMacdonaldBlack Conflicts

**Comment by CalebMacdonaldBlack** [comment]
@axrs

**Comment by CalebMacdonaldBlack** [comment]
@axrs

**Review by axrs — APPROVED** [review]

**Pull request merged**